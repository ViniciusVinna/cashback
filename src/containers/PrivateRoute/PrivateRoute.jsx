import React from 'react';
import PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router-dom';

const PrivateRoute = ({ children, ...rest }) => {
  const isLogged = true;

  if (isLogged) {
    return (
      <Route {...rest}>
        {children}
      </Route>
    );
  }

  return (<Redirect to="/signin" />);
};

PrivateRoute.propTypes = {
  children: PropTypes.node,
};

export default PrivateRoute;
