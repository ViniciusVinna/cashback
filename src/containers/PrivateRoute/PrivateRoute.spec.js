import React from 'react';
import { useSelector } from 'react-redux';

import PrivateRoute from './PrivateRoute';

let mockStore = {
  user: {
    isLogged: false,
    accessToken: '',
  },
};

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('PrivateRoute', () => {
  beforeEach(() => {
    useSelector.mockImplementation(callback => callback(mockStore));
  });

  afterEach(() => {
    useSelector.mockClear();
  });

  it('it should redirect to /', () => {
    const { container } = render(
      <PrivateRoute path="/dashboard">
        <div>Dashboard Content</div>
      </PrivateRoute>
    );
    expect(container).toMatchSnapshot();
  });

  it('it should redirect to Dashboard when user is logged and has an access token', () => {
    mockStore = {
      user: {
        isLogged: true,
        accessToken: true,
      },
    };

    const { container } = render(
      <PrivateRoute path="/dashboard">
        <div>Dashboard Content</div>
      </PrivateRoute>
    );
    expect(container).toMatchSnapshot();
  });
});
