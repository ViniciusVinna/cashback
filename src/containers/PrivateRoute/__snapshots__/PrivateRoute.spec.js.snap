// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute it should redirect to / 1`] = `<div />`;

exports[`PrivateRoute it should redirect to Dashboard when user is logged and has an access token 1`] = `<div />`;
